modules/smtp/site.js

```diff
@@ -97,6 +97,20 @@ var smtp_delete_action = function(event) {
     );
 };
 
+var error_if_there_is_leading_or_trailing_space = function(element) {
+    if (element.val() !== element.val().trim()) {
+        if (element.attr('name').slice(-4) === 'user') {
```

Comment:
@NasserNgandu I think we cannot rely on user in the field name. In nux module username has nux_email as name same as for password.

Reply:
According to me this function could receive the text and the type of the input (username or password). Then should check if the text contains leading/trailing space and according to the type build the error message. Then each module which has a username/password (smtp, imap, nux) should add events on their own inputs and call this methods for validation.